pkg/config/remote/service/client_predicates.go

```diff
@@ -112,15 +111,14 @@ func executePredicate(client *pbgo.Client, predicates []*pbgo.TracerPredicateV1)
 				}
 
 				matched, errs := versionConstraint.Validate(version)
-				if len(errs) != 0 {
-					return false, fmt.Errorf("errors: %s", errs)
-				}
+				// We don't return on error here, it's simply the version not matching
 				if !matched || len(errs) > 0 {
-					return false, nil
+					continue
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Can you add a test for this path?

Reply:
There is one [here](https://github.com/DataDog/datadog-agent/pull/12980/files#diff-4ff734e21cc66f971c7e2118dee491340e10c87f708d828dfd1306a4a602a0eeR82) already